specreduce/extract.py

```diff
@@ -3,216 +3,162 @@
-            mask=mask
-        )
-        skyspec = Spectrum1D(
-            spectral_axis=np.arange(len(onedspec)) * u.pixel,
-            flux=skysubflux * img_unit,
-            mask=mask
-        )
-
-        return spec, skyspec
-
-    def _extract_from_box(self, image, wave_index, low_end, high_end, extracted_result):
-
-        # compute nearest integer endpoints defining an internal interval,
-        # and fractional pixel areas that remain outside this interval.
-        # (taken from the HST STIS pipeline code:
-        # https://github.com/spacetelescope/hstcal/blob/master/pkg/stis/calstis/cs6/x1dspec.c)
-        #
-        # This assumes that the pixel coordinates represent the center of the pixel.
-        # E.g. pixel at y=15.0 covers the image from y=14.5 to y=15.5
-
-        # nearest integer endpoints
-        j1 = self._find_nearest_int(low_end)
-        j2 = self._find_nearest_int(high_end)
-
-        # fractional pixel areas at the end points
-        s1 = 0.5 - (low_end - j1)
-        s2 = 0.5 + high_end - j2
-
-        # add up the total flux around the trace_line
-        extracted_result[wave_index] = np.nansum(image[j1 + 1:j2, wave_index])
-        extracted_result[wave_index] += np.nansum(image[j1, wave_index]) * s1
-        extracted_result[wave_index] += np.nansum(image[j2, wave_index]) * s2
-
-    def _find_nearest_int(self, end_point):
-        if (end_point % 1) < 0.5:
-            return int(end_point)
-        else:
-            return int(end_point + 1)
-
-    def get_checkplot(self):
-        trace_line = self.last_trace.line
-
-        fig = plt.figure()
-        plt.imshow(self.last_img, origin='lower', aspect='auto', cmap=plt.cm.Greys_r)
-        plt.clim(np.percentile(self.last_img, (5, 98)))
-
-        plt.plot(np.arange(len(trace_line)), trace_line, c='C0')
-        plt.fill_between(
-            np.arange(len(trace_line)),
-            trace_line + self.apwidth,
-            trace_line - self.apwidth,
-            color='C0',
-            alpha=0.5
-        )
-        plt.fill_between(
-            np.arange(len(trace_line)),
-            trace_line + self.apwidth + self.skysep,
-            trace_line + self.apwidth + self.skysep + self.skywidth,
-            color='C1',
-            alpha=0.5
-        )
-        plt.fill_between(
-            np.arange(len(trace_line)),
-            trace_line - self.apwidth - self.skysep,
-            trace_line - self.apwidth - self.skysep - self.skywidth,
-            color='C1',
-            alpha=0.5
-        )
-        plt.ylim(
-            np.min(
-                trace_line - (self.apwidth + self.skysep + self.skywidth) * 2
-            ),
-            np.max(
-                trace_line + (self.apwidth + self.skysep + self.skywidth) * 2
-            )
-        )
-
-        return fig
+    # TODO: what is a reasonable default?
+    # TODO: int or float?
+    width: int = 5
+
+    # TODO: should disp_axis and crossdisp_axis be defined in the Trace object?
```

Comment:
My personal opinion would be to either put all options (except maybe `image` and `trace_object` either in the class itself or as kwargs, but mixing the two will probably cause confusion.

Reply:
I personally think the axis information should go in the `Trace` instance. That however requires changes in `tracing.py` to allow it to handle both cases of 2D spectra.

Reply:
I think adding support for WCS is coming soon in a separate ticket. A properly set spectral WCS instance renders `disp_axis` and `crossdisp_axis` redundant. Maybe we should leave the extraction API as is for now, and work the axis issue when implementing WCS? 
In support to @tepickering 's argument, I think  extractions on "transposed" spectral images was deemed low-priority at some point.

Reply:
we need to be a bit careful how deeply we go down the WCS rabbit hole and where. a proper spectral WCS requires a wavelength calibration which may or may not be available at the extraction step. in fact, in a common use case, the aperture used for extraction is then applied to a arc lamp image to extract a spectrum that is then used to determine the wavelength calibration. using WCS information if its there is fine and would be very handy for re-analysis use cases, but we can't assume it will always be there.

Reply:
It looks like all options have now been moved from the class to arguments to `__call__` resolving my original comment and the further discussion about moving axis information to `Trace` is captured in #83.  Are we safe to mark this conversation as resolved for the sake of getting this PR approved?

Reply:
From my viewpoint, it can be marked as resolved.